analyzers/Directory.Build.props

```diff
@@ -24,4 +24,13 @@
     <DisableImplicitNuGetFallbackFolder>true</DisableImplicitNuGetFallbackFolder>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(TF_BUILD)' == 'true'">
+    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
+    <Deterministic>true</Deterministic>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <SourceRoot Include="$(MSBuildThisFileDirectory)/"/>
```

Comment:
Needed when `ContinuousIntegrationBuild` and `Deterministic` are set to true. Otherwise we get the following error:
```
Microsoft.Managed.Core.targets(191,5): error : SourceRoot items must include at least one top-level (not nested) item when DeterministicSourcePaths is true
```
We are using `MSBuildThisFileDirectory` as recommended here: https://github.com/dotnet/roslyn/issues/37379#issuecomment-513371985